torch/distributed/_tensor/ops/pointwise_ops.py

```diff
@@ -382,19 +382,9 @@ def _register_non_deterministic_op(op):
     def non_deterministic_rule(op_schema: OpSchema) -> OutputSharding:
         self_spec = cast(DTensorSpec, op_schema.args_schema[0])
 
-        # TODO: We are specializing non_deterministic_rule now because
-        # replicate does not support this op yet. We should remove
-        # this rule once we support non-deterministic op for replicate.
-        replicate_or_partial = False
-        for placement in self_spec.placements:
-            if isinstance(placement, (Replicate, _Partial)):
-                replicate_or_partial = True
-                break
-
-        if replicate_or_partial:
-            return OutputSharding(
-                None, failed_reason=f"{op} with replication is not supported yet!"
-            )
+        # TODO: we will support native_dropout in future
+        if op == aten.native_dropout.default:
```

Comment:
why we remove it as a whole? We used to support the dropout with shard placement

Reply:
You're right. Even after using the new RNG approach, `dropout` with `_Partial` is still not supported so the current logic should be kept.